tests/ui/cast/cast-rfc0401-vtable-kinds.rs

```diff
@@ -17,7 +17,7 @@ impl Foo<u32> for u32 { fn foo(&self, _: u32) -> u32 { self+43 } }
 impl Bar for () {}
 
 unsafe fn round_trip_and_call<'a>(t: *const (dyn Foo<u32>+'a)) -> u32 {
-    let foo_e : *const dyn Foo<u16> = t as *const _;
+    let foo_e : *const dyn Foo<u32> = t as *const _;
```

Comment:
Not sure if this intentionally changed the type or if it was intended to only cast away the lifetime. But we probably shouldn't have ui tests with UB. ;)